### ISSUE TITLE
Open Refine version 3.1 merge columns not working with null values

### ISSUE THREAD

**Issue by jenyoung** [OpenRefine/OpenRefine#1841]
**Describe the bug**
I want to merge two columns together, both contain `null` values.

According to [this documentation](https://github.com/OpenRefine/OpenRefine/wiki/Recipes#7-columns), starting with 3.0 I should be able to combine them without having to change the values to an `empty string` first. That is not the case for 3.1-beta

**Expected behavior**
I expected to be able to merge my columns regardless of `null` values.

**Screenshots**
Not working with `null` values

![2018-11-20_154923](https://user-images.githubusercontent.com/2250501/48807377-fa3f0c00-ece2-11e8-9e6f-e18d776d6dae.png)

Working after changing the `nulls` to `empty strings`

![2018-11-20_160818](https://user-images.githubusercontent.com/2250501/48807412-0cb94580-ece3-11e8-9fdb-c834127347b7.png)

**Desktop (please complete the following information):**
 - OS: `Windows 10, Version 1703, OS Build 15063.1387`
 - Browser Version: `Version 69.0.3497.100 (Official Build) (64-bit)`
 - JRE: ` JRE 1.8.0_181`

**OpenRefine (please complete the following information):**
 - Version` OpenRefine 3.1 Beta`

**Comment by ostephens** [comment]
It looks like the issue here is that currently:

`"string"+null -> null`

Rather than

`"string"+null -> "string"`

If you use `toString()` on the `null` then the `null` gets converted to an empty string and thus you get the desired behaviour:

`"string"+null.toString() -> "string"

I'm not sure when the `"string"+null -> null` behaviour started, as in #1635 I documented that 

`"example"+null -> "example"`

So this seems to be something that has changed since June.

I'll investigate further

**Comment by ostephens** [comment]
It looks like I was incorrect when I documented the behaviour of `"string"+null` in #1635. Testing back to 2.7 this results in `null`, and the code that controls this hasn't changed since 2011

I'm slightly unsure what the "right" behaviour should be when you attempt to add `null` to:

* A string
* A number
* A `null`
* A date

Happy to make changes here but we could do with agreeing behaviour

@jenyoung @wetneb @ettorerizza @thadguidry  (feel free to add others)

**Comment by thadguidry** [comment]
For the case of concatenation in GREL...
I would expect users want to have more power and a say so in how null is treated in an expression.
Which is also why we have Facet option now that treats blanks and nulls individually.

I think most uses in the case of concatenation in GREL expect that null should be treated as a blank.
But, we could certainly give THEM the choice, and simply introduce a checkbox on the Expression dialog that says "treat null as blank" , that way we don't break existing handling.

**Comment by jenyoung** [comment]
Yes please! I had already gone through my data and changed `empty strings` to `null` and to have to change it back when I just want to merge or concatenate columns is not ideal.

**Comment by ostephens** [comment]
@jenyoung just to make sure you know - you don't have to change the nulls to empty strings in the project overall, just use GREL like

`value.toString() + cells["worktype[1]'].value.toString()` 

By adding `toString()` that will make sure any null values are treated as empty strings when concatenating.

Aside from this it seems like the solution is to have a `concatenate columns` function as per #1473
